### ISSUE TITLE
TomcatStarter not logging stacktrace.

### ISSUE THREAD

**Issue by robrechta** [spring-projects/spring-boot#12154]
The tomcat starter is not logging the stacktrace, which causes a long search when a spring-bean cannot be created.
TomcatStarter --> `org.springframework.boot.web.embedded.tomcat.TomcatStarter`
This is the current situation (spring-boot-2.0.0.RC1)
```
catch (Exception ex) {
	this.startUpException = ex;
	// Prevent Tomcat from logging and re-throwing when we know we can
	// deal with it in the main thread, but log for information here.
	if (logger.isErrorEnabled()) {
		logger.error("Error starting Tomcat context. Exception: "
				+ ex.getClass().getName() + ". Message: " + ex.getMessage());
	}
}
```
Enhancement:
```
catch (Exception ex) {
	this.startUpException = ex;
	// Prevent Tomcat from logging and re-throwing when we know we can
	// deal with it in the main thread, but log for information here.
	if (logger.isErrorEnabled()) {
		logger.error("Error starting Tomcat context. Exception: "
				+ ex.getClass().getName() + ". Message: " + ex.getMessage(), ex);
	}
}
```
Then the stacktrace is printed too

**Comment by wilkinsona** [comment]
In my experience, the stack trace is always logged elsewhere. Can you provide a sample that shows the stack trace not being logged at all?

**Comment by robrechta** [comment]
I have to create a small project with that issue, will take some time. But I can explain what error I had.
I'm working with a spring-boot app together with kotlin, and I'm binding custom properties to a property-class. I made a mistake in the property class, and didn't found directly what the problem was. I had to debug to see the real cause.
An example of the error I made

Property-file
```
myapp.resoure=something
```
Propery-object
```
@ConfigurationProperties(prefix = "myapp")
open class MyAppProperties(
        val resource: String,
)
```
The error was that the resource property in the MyAppProperties must be defined with var, and not val. The exception was thrown in `org.springframework.boot.context.properties.ConfigurationPropertiesBinder` but never fully logged. Only the message appeared `Could not bind properties to ...MyAppProperties ....`  but no information of what the exact problem was given. So I didn't know directly to look at that property. 
In the original cause, the property is mentioned, so I know where to look, and saw immediate that val was incorrect. I saw the original cause only when debugging in the `TomcatStarter`
